### PR TITLE
expose the PodCIDR in node info

### PR DESCRIPTION
**Pull request by jacobstr**
**What this PR does / why we need it**:

It can be helpful to know the PodCIDR of nodes in order to predict IP
space utilization. In GKE for example, one typically uses a Secondary
Alias IP range that doles out PodCIDRs dynamically. When this is
exhausted, nodes will fail to boot with messages such as:

```
Instance 'gke-test-ingre-43fd70d6-9v77' creation failed: IP space of 'projects/network-project/regions/us-central1/subnetworks/pod-alias-b85f82b0bca36afa' is exhausted.
```

Although not comprehensive, on could use this label to construct a query
such as:

```
count(kube_node_info(pod_cidr~="/24")) * 256
```

### CODE REVIEW

File: internal/store/node.go [hunk 1]
```diff
@@ -59,6 +60,7 @@ var (
 								n.Status.NodeInfo.KubeletVersion,
 								n.Status.NodeInfo.KubeProxyVersion,
 								n.Spec.ProviderID,
+								n.Spec.PodCIDR,
```
Comment (LiliC) [inline internal/store/node.go]: This is an optional label, not sure how we usually handle optional labels cc @brancz @tariq1890

File: internal/store/node_test.go [hunk 2]
```diff
@@ -46,6 +46,7 @@ func TestNodeStore(t *testing.T) {
 				},
 				Spec: v1.NodeSpec{
 					ProviderID: "provider://i-uniqueid",
+					PodCIDR:    "172.24.10.0/24",
```
Comment (LiliC) [inline internal/store/node_test.go]: Since this can be empty or not set, lets have one test where this is not set.

File: internal/store/node.go (hunk 1 above)
Comment (brancz) [reply internal/store/node.go]: It’s not a pointer so at worst this is the empty string in which case Prometheus just drops the label so it’s fine as is.

File: internal/store/node_test.go (hunk 2 above)
Comment (jacobstr) [reply internal/store/node_test.go]: I'm adding a general test case with an empty `v1.Node` and verifying all empty strings. Not sure if you have preferences over the ordering of test cases but I'm happy to change that.

**Comment by jacobstr** [comment]
Note: I'm not committing a spurious ```
+	github.com/gogo/protobuf v1.1.1 // indirect
```

That show up in my local diff. I don't expect any changes to `go.mod` contents given I have not changed added / updated deps.

**Comment by brancz** [comment]
Thanks!

/lgtm
/approve
